src/Languages/JavaScript.ts

```diff
@@ -0,0 +1,415 @@
+import { EnumProperties } from "./Properties/EnumProperties";
+import { ExceptionProperties } from "./Properties/ExceptionProperties";
+import { FunctionProperties } from "./Properties/FunctionProperties";
+import { GeneralProperties } from "./Properties/GeneralProperties";
+import { ImportProperties } from "./Properties/ImportProperties";
+import { InterfaceProperties } from "./Properties/InterfaceProperties";
+import { LambdaProperties } from "./Properties/LambdaProperties";
+import { ListProperties } from "./Properties/ListProperties";
+import { LoopProperties } from "./Properties/LoopProperties";
+import { MathProperties } from "./Properties/MathProperties";
+import { NativeCallProperties, NativeCallScope, NativeCallType } from "./Properties/NativeCallProperties";
+import { NumberProperties } from "./Properties/NumberProperties";
+import { OperatorProperties } from "./Properties/OperatorProperties";
+import { OutputProperties } from "./Properties/OutputProperties";
+import { ParameterProperties } from "./Properties/ParameterProperties";
+import { StringProperties } from "./Properties/StringProperties";
+import { StringFormatProperties } from "./Properties/StringFormatProperties";
+import { StyleProperties } from "./Properties/StyleProperties";
+import { VariableProperties } from "./Properties/VariableProperties";
+
+/**
+ * A summary of information for the JavaScript language.
+ */
+export class JavaScript extends CLikeLanguage {
+    /**
+     * Generates metadata on arrays.
+     * 
+     * @param arrays   A property container for metadata on arrays.
+     */
+    protected generateArrayProperties(arrays: ArrayProperties): void {
+        arrays.className = "Array";
+        arrays.initializeAsNew = false;
+        arrays.initializeViaStatic = true;
+        arrays.length = new NativeCallProperties(
+            "length",
+            NativeCallScope.Member,
+            NativeCallType.Property);
+    }
+
+    /**
+     * Generates metadata on booleans.
+     * 
+     * @param booleans   A property container for metadata on booleans.
+     */
+    protected generateBooleanProperties(booleans: BooleanProperties): void {
+        booleans.className = "";
+    }
+
+    /**
+     * Generates metadata on classes.
+     * 
+     * @param classes   A property container for metadata on classes.
+     */
+    protected generateClassProperties(classes: ClassProperties): void {
+        super.generateClassProperties(classes);
+
+        classes.aliases = {
+            "dictionary": "object",
+            "double": "number",
+            "float": "number",
+            "int": "number"
+        };
+        classes.constructorKeyword = "constructor";
+        classes.constructorUsesKeyword = true;
+        classes.declareExtendsLeft = " extends ";
+        classes.declareImplementsLeft = " implements ";
+        classes.declareStartRight = " {";
+        classes.superConstructor = "super";
+    }

```

Comment:
If the base class is saying generics are supported, you should change it to not specify, and then child classes to individually specify.